src/predict.jl

```diff
@@ -0,0 +1,292 @@
+a subset of the grouping variables to use, but not at this time.
+
+For `GeneralizedLinearMixedModel`, the `type` parameter specifies
+whether the predictions should be returned on the scale of linear predictor
+(`:linpred`) or on the response scale (`:response`). If you don't know the
+difference between these terms, then you probably want `type=:response`.
+
+!!! warning
+    Models are assumed to be full rank.
+
+!!! note
+    The `predict` and `predict!` methods with `newX` as a fixed effects matrix
+    differ from the methods with `newdata` in tabular format. The matrix
+    methods are more efficient than the tabular methods. The tabular methods
+    can accomodate new values of the grouping variable(s), but the matrix methods
+    cannot.
+"""
+function StatsBase.predict!(y::AbstractVector{<:Union{T, Missing}}, m::LinearMixedModel{T}, newX::AbstractMatrix{T}=m.X;
+                           use_re=true) where T
+    # this is called `use_re` in case we later decide to support prediction
+    # with only a subset of the RE
+
+    mul!(y, newX, m.β)
+
+    use_re || return y
+
+    # because we're using ranef(), this actually
+    # adds the *scaled* random effects
+    for (rt, bb) in zip(m.reterms, ranef(m))
+        unscaledre!(y, rt, bb)
+    end
+
+    y
+end
+
+function StatsBase.predict!(y::AbstractVector{<:Union{T, Missing}}, m::GeneralizedLinearMixedModel{T}, newX::AbstractMatrix{T}=m.X;
+                            use_re=true, type=:response) where T
+    # this is called `use_re` in case we later decide to support prediction
+    # with only a subset of the RE
+
+    type in (:linpred, :response) || throw(ArgumentError("Invalid value for `type`: $(type)"))
+
+    mul!(y, newX, m.β)
+
+    if use_re
+        # because we're using ranef(), this actually
+        # adds the *scaled* random effects
+        for (rt, bb) in zip(m.reterms, ranef(m))
+            unscaledre!(y, rt, bb)
+        end
+    end
+
+    type == :linpred && return y
+
+    @inbounds for (idx, val) in enumerate(y)
+        y[idx] = linkinv(Link(m.resp), val)
+    end
+
+    y
+end
+
+function StatsBase.predict(m::LinearMixedModel{T}, newX::AbstractMatrix{T}=m.X;
+                          use_re=true) where T
+    # this is called `use_re` in case we later decide to support prediction
+    # with only a subset of the RE
+
+    y = zeros(T, nobs(m))
+    predict!(y, m, newX; use_re=use_re)
+end
+
+function StatsBase.predict(m::GeneralizedLinearMixedModel{T}, newX::AbstractMatrix{T}=m.X;
+                           use_re=true, type=:response) where T
+    # this is called `use_re` in case we later decide to support prediction
+    # with only a subset of the RE
+
+    type in (:linpred, :response) || throw(ArgumentError("Invalid value for type: $(type)"))
+
+    if use_re && newX === m.X
+        # should we add a kwarg to fitted to allow returning eta?
+        type == :response && return fitted(m)
+        return m.resp.eta
```

Comment:
I would probably write those two lines as 
```suggestion
        return type == : response ? fitted(m) : m.resp.eta
```

Reply:
yep, definitely better :)